packages/firebase_admob/android/src/main/java/io/flutter/plugins/firebaseadmob/FirebaseAdMobPlugin.java

```diff
@@ -17,29 +22,49 @@
 import java.util.Locale;
 import java.util.Map;
 
-public class FirebaseAdMobPlugin implements MethodCallHandler {
-
-  private final Registrar registrar;
-  private final MethodChannel channel;
-
-  RewardedVideoAdWrapper rewardedWrapper;
-
+/**
+ * Flutter plugin accessing Firebase Admob API.
+ *
+ * <p>Instantiate this in an add to app scenario to gracefully handle activity and context changes.
+ */
+public class FirebaseAdMobPlugin implements FlutterPlugin, ActivityAware, MethodCallHandler {
+  private Context applicationContext;
+  private MethodChannel channel;
+  private Activity activity;
+  private FlutterPluginBinding pluginBinding;
```

Comment:
Add some code comments for which instance variable is available for which plugin APIs (and warn future maintainers about which ones may be null)